ecosystem/indexer/src/indexer/fetcher.rs

```diff
@@ -200,11 +192,22 @@ impl TransactionFetcherTrait for TransactionFetcher {
     }
 
     async fn fetch_ledger_info(&mut self) -> State {
-        self.client
-            .get_ledger_information()
-            .await
-            .expect("ledger info must be present")
-            .into_inner()
+        let mut retries = 0;
+        while retries < MAX_RETRIES {
+            retries += 1;
+            match self.client.get_ledger_information().await {
+                Ok(inner) => return inner.into_inner(),
+                Err(err) => {
+                    aptos_logger::error!(
+                        "Failed to get ledger info, will retry in {}ms. Err: {:?}",
+                        RETRY_TIME_MILLIS,
+                        err
+                    );
+                    tokio::time::sleep(Duration::from_secs(RETRY_TIME_MILLIS)).await
+                }
+            }
+        }
+        panic!("Failed to get ledger info.");
```

Comment:
Let's talk about this, we shouldn't have panics unless there are unforseen errors, since it should loop forever (it's just an indexer)

Reply:
Ah I see the argument. So this function is only used to check the chain id which runs once when we start the server. The thinking is that if the API is down or the query has some issues (e.g. we're on an incompatible version of the indexer), we should just fail and have the user fix something instead of pretending to start the indexer. 

We make the same get_ledger_information() call on a loop separately as well and I agree that we should keep the forever loop there